--- a/src/configuration/Registry.cpp
+++ b/src/configuration/Registry.cpp
@@ -87,7 +87,7 @@
 
 	for (int i = 0; i < s.getLength(); i++) {
 
-		std::string tmp = s[i]["name"];
+		std::string tmp = s[i]["name"].c_str();
 		if (tmp == objname)
 			return s[i];
 	}
--- a/src/solarpowerlog.cpp
+++ b/src/solarpowerlog.cpp
@@ -145,7 +145,7 @@
 	}
 
 	try {
-		std::string s = set;
+		std::string s = set.c_str();
 		cout << "= " << s << "\t is";
 	} catch (...) {
 
